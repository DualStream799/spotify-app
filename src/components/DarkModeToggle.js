import React from 'react';
import useDarkMode from 'use-dark-mode';
import Toggle from './Toggle';

const DarkModeToggle = () => {
  const darkMode = useDarkMode(false);

  return (
    <div>
      <button type="button" onClick={darkMode.enable}>
        Dark Mode
      </button>
      <Toggle checked={darkMode.value} onChange={darkMode.toggle} />
    </div>
  );
};

export default DarkModeToggle;