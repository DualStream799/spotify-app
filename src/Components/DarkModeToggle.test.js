import React from 'react';
import DarkModeToggle from './DarkModeToggle';
import renderer from 'react-test-renderer';

describe('Dark Mode Button ', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <DarkModeToggle>
        Dark Mode
      </DarkModeToggle>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});