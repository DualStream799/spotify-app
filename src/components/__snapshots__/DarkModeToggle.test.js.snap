// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark Mode Button  renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Dark Mode
  </button>
  <span
    className="toggle-control"
  >
    <input
      checked={false}
      className="dmcheck"
      id="dmcheck"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="dmcheck"
    />
  </span>
</div>
`;
